### PR TITLE
Rate Limit Kuru Squash

### LINKED ISSUE #3: Implement in-site ratelimiter
Current problems is:

- Autoclickers can still abuse the button (can be patched with a timer?)
- Might need to do secure headers for added security.

### PR DESCRIPTION
**Pull request by Amnish04**
Closes #3 

@sr229 
This PR:

1. Removes the **Squash** button from DOM, after **THRESHOLD** clicks
2. Displays a message suggesting how much time left until more squashes available.
3. Brings back the button after wait time expires.

We can tweak the interval time and threshold clicks as desired.

Please let me know if we need changes

### CODE REVIEW

File: islands/CounterCard.tsx [hunk 1]
```diff
@@ -51,8 +51,33 @@ export default function Counter(props: SharedProps) {
   const [timer, setTimer] = useState(0);
   const ipc = new BroadcastChannel("counterIpc");
 
+  const THRESHOLD_CLICKS = 10; // Maximum number of clicks in an interval
```
Comment (sr229) [inline islands/CounterCard.tsx]: I think its better if we do it every 25 or 50 clicks so it doesn't feel too strict

**Review by sr229 — CHANGES_REQUESTED** [review]
Love the changes so far, but we can make it better.

File: islands/CounterCard.tsx (hunk 1 above)
Comment (Amnish04) [reply islands/CounterCard.tsx]: Just changed the value to `30`

**Comment by Amnish04** [comment]
@sr229 
Just improved the timeout algorithm. Previously, the timer kept running in the background due to which a full timeout of 30 seconds was never given to the user. That problem should be fixed now!

**Review by sr229 — APPROVED** [review]

**Comment by sr229** [comment]
Thanks for your contribution, if you're participating for Hacktoberfest, this should count towards your PR 🎉

**Comment by Amnish04** [comment]
@sr229 It was my pleasure contributing! 
Just one request, could you please add the `hacktoberfest-accepted` label to the **PR** as well. I know this PR  counts anyways, but just trying to avoid weird hacktoberfest glitches that have happened in the past.

Thank you :D

**Comment by sr229** [comment]
> @sr229 It was my pleasure contributing! 
> Just one request, could you please add the `hacktoberfest-accepted` label to the **PR** as well. I know this PR  counts anyways, but just trying to avoid weird hacktoberfest glitches that have happened in the past.
> 
> Thank you :D

Sure thing, give me an hour and I'll process it, thanks again!

**Pull request merged**